index.js

```diff
@@ -28,11 +30,37 @@ module.exports = function (str, cols) {
 		}
 
 		if (++visible >= cols - 2 && lastSpace > 0) {
-			ret = spliceString(ret, lastSpace, 1, '\n');
+			pre = spliceString(pre, lastSpace, 1, '\n');
 			lastSpace = 0;
 			visible = 0;
 		}
 	}
 
+	visible = 0;
+	for (var j = 0; j < pre.length; j++) {
+		var y = pre[j];
+		ret += y;
+		if (y === '\u001b') {
+			var code = parseFloat(pre.slice(j, j + 4).match(/[0-9][^m]*/));
+			escapeCode = (code === 39) ? undefined : code;
+		} else if (insideEscape && y === 'm') {
+			insideEscape = false;
+			continue;
+		}
+
+		if (insideEscape) {
+			continue;
+		}
+
+		++visible;
```

Comment:
No point in using a prefix increment when you don't use the value directy: `++visible;` => `visible++;`.